src/unpacker/unpack_base.py

```diff
@@ -20,15 +21,16 @@ def extract_files_from_file(self, file_path, tmp_dir):
         self._initialize_shared_folder(tmp_dir)
         shutil.copy2(file_path, str(Path(tmp_dir, 'input', Path(file_path).name)))
 
-        output, return_code = execute_shell_command_get_return_code('docker run --privileged -v /dev:/dev -v {}:/tmp/extractor --rm fkiecad/fact_extractor'.format(tmp_dir))
+        output, return_code = execute_shell_command_get_return_code(
+            'docker run --privileged -m {}m -v /dev:/dev -v {}:/tmp/extractor --rm fkiecad/fact_extractor'.format(self.config.get('unpack', 'memory_limit', fallback='1024'), tmp_dir)
+        )
         if return_code != 0:
             error = 'Failed to execute docker extractor with code {}:\n{}'.format(return_code, output)
             logging.error(error)
             raise RuntimeError(error)
 
         self.change_owner_back_to_me(tmp_dir)
-        all_items = list(Path(tmp_dir, 'files').glob('**/*'))
-        return [item for item in all_items if not item.is_dir()]
+        return [item for item in safe_rglob(Path(tmp_dir, 'files')) if not item.is_dir()]
 
     def change_owner_back_to_me(self, directory: str = None, permissions='u+r'):
         with Popen('sudo chown -R {}:{} {}'.format(getuid(), getgid(), directory), shell=True, stdout=PIPE, stderr=PIPE) as pl:
```

Comment:
Could you change this to use execute_shell_command helper function. This command does not work correctly, since any output on stderr overwrites stdout.